cpp/oneapi/dal/backend/primitives/lapack/solve_dpc.cpp

```diff
@@ -59,6 +61,167 @@ inline sycl::event beta_copy_transform(sycl::queue& queue,
     });
 }
 
+/*
+This is an adaptation of the CPU version, which can be found in this file:
+cpp/daal/src/algorithms/service_kernel_math.h
+
+It solves a linear system A*x = b
+where 'b' might be a matrix (multiple RHS)
+
+It is intended as a fallback for solving linear regression, where these
+matrices are obtained as follows:
+    A = t(X)*X
+    b = t(X)*y
+
+It can handle systems that are not positive semi-definite, so it's used
+as a fallback when Cholesky fails or when it involves too small numbers
+(which tends to happen when floating point error results in a slightly
+positive matrix when it should have zero determinant in theory).
+*/
+template <mkl::uplo uplo, typename Float>
+sycl::event solve_spectral_decomposition(
+    sycl::queue& queue,
+    ndview<Float, 2>& A, // t(X)*X, LHS, gets overwritten
+    sycl::event& event_A,
+    ndview<Float, 2>& b, // t(X)*y, RHS, solution is outputted here
+    sycl::event& event_b,
+    const std::int64_t dim_A,
+    const std::int64_t nrhs) {
+    constexpr auto alloc = sycl::usm::alloc::device;
+
+    /* Decompose: A = Q * diag(l) * t(Q) */
+    /* Note: for NRHS>1, this will overallocate in order to reuse the memory as buffer later on */
+    auto eigenvalues = array<Float>::empty(queue, dim_A * nrhs, alloc);
+    auto eigenvalues_view = ndview<Float, 1>::wrap(eigenvalues);
```

Comment:
Does it make sense to replace ndview and array with ndarray?

Reply:
There is magic conversion that works:
https://github.com/oneapi-src/oneDAL/blob/main/cpp/oneapi/dal/algo/pca/backend/gpu/misc.hpp#L77
here ndarray has been provided in syevd

Reply:
Thanks. And what would be the difference between `array` and `ndarray`? Looks like they use the same allocator.

Reply:
In general would be no difference, but you can avoid allocation of ndview

Reply:
Changed to ndarray.